packages/dds/tree/src/test/shared-tree/treeApi.spec.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "node:assert";
+import { Tree } from "../../shared-tree/index.js";
+import { SchemaFactory, TreeConfiguration } from "../../simple-tree/index.js";
+import { getView } from "../utils.js";
+
+const schema = new SchemaFactory("com.example");
+class TestObject extends schema.object("TestObject", { content: schema.number }) {}
+
+describe("treeApi", () => {
+	describe("runTransaction invoked via a tree view", () => {
+		it("runs transactions", () => {
+			const view = getView(new TreeConfiguration(TestObject, () => ({ content: 42 })));
+			Tree.runTransaction(view, (root) => {
+				root.content = 43;
+			});
+			assert.equal(view.root.content, 43);
+		});
+
+		it("can be rolled back", () => {
+			const view = getView(new TreeConfiguration(TestObject, () => ({ content: 42 })));
```

Comment:
for the rollback tests, would be good to ensure we get the events for state being rolled back too

Reply:
Added tests